src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -110,8 +139,14 @@ public void VerifyResolvedBindings(FunctionMetadata functionMetadata, IEnumerabl
 
             ApplyMethodLevelAttributes(functionMetadata, triggerMetadata, methodAttributes);
 
+            Type parameterType = null;
+            if (BindingAttributeContainsExpression(inputBindings.Union(outputBindings)))
+            {
+                parameterType = typeof(ParameterBindingData);
```

Comment:
since we're setting `parameterType` here, we don't ever go through the GetRequestedType path